lightcurve/alcdef.py

```diff
@@ -0,0 +1,84 @@
+from django.contrib import admin
+from django.db import models
+from django.http import HttpResponse
+
+class AlcdefWriter(models.Model):
+    def __init__(self):
+        self.metadata = {
+            'REVISEDDATA': '',
+            'OBJECTNUMBER': '',
+            'OBJECTNAME': '',
+            'MPCDESIG': '',
+            'CONTACTNAME': '',
+            'CONTACTINFO': '',
+            'OBSERVERS': '',
+            'OBSLONGITUDE': '',
+            'OBSLATITUDE': '',
+            'SESSIONDATE': '',
+            'SESSIONTIME': '',
+            'FILTER': '',
+            'MAGBAND': '',
+            'STANDARD': '',
+            'DIFFERMAGS': '',
+            'LTCTYPE': '',
+            'LTCDAYS': '',
+            'LTCAPP': '',
+            'REDUCEDMAGS': '',
+            'UCORMAG': '',
+            'OBJECTRA': '',
+            'OBJECTDEC': '',
+            'PHASE': '',
+            'PABL': '',
+            'PABB': '',
+            'CICORRECTION': '',
+            'CIBAND': '',
+            'CITARGET': '',
+            'PUBLICATION': '',
+            'BIBCODE': '',
+            'DELIMITER': '',
+            'COMMENT': '',
+        }
+        self.starid = 1 # ID for metadata of comparison star
+        self.comments = []
+        self.datas = []
+
+    def set(self, key, value):
+        self.metadata[key] = value
```

Comment:
This would allow the user to set *any* key despite the whitelist above.

I would move the whitelist to a constant outside of the class, eg. `ALLOWED_KEYWORDS`.  Make it a `set` data type and you can easily check it like this:

```
def set(self, key, value):
    if ALLOWED_KEYWORDS.has(key):
        self.metadata[key] = value
```

This means `self.metadata = {}` to start and you won't write a bunch of empty keywords if the user didn't set them.

You may want to have a separate list of `REQUIRED_KEYWORDS` that you check to make sure are included before writing.

Reply:
Done.